composer/algorithms/ghost_batchnorm/ghost_batchnorm.py

```diff
@@ -155,36 +153,49 @@ def __init__(self, base_batchnorm: _TORCH_BATCHNORM_BASE_CLASS, ghost_batch_size
         super().__init__()
         self.ghost_batch_size = ghost_batch_size
         self.batchnorm = base_batchnorm
+        self.current_nchunks: Optional[int] = None
 
     def _has_momentum(self) -> bool:
         return hasattr(self.batchnorm, 'momentum') and self.batchnorm.momentum is not None
 
     def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
         batch_size = input.shape[0]
+
         if batch_size < self.ghost_batch_size:
             raise ValueError(f"Worker batch size {batch_size} < ghost_batch_size {self.ghost_batch_size}")
 
-        nchunks = int(np.ceil(batch_size / self.ghost_batch_size))
-        has_momentum = self._has_momentum()
-        if has_momentum:
+        nchunks: int = int(math.ceil(batch_size / self.ghost_batch_size))
+        scale = self.batchnorm.momentum is not None
+        original_momentum: float = self.batchnorm.momentum
+
+        if self.training and scale:
             # applying the same batchnorm multiple times greatly increases
             # the variance of the moving average statistics; reduce the
             # exponential moving average constant proportionally
-            # to partially compensate for this
-            original_momentum = self.batchnorm.momentum
-            self.batchnorm.momentum = float(original_momentum) / nchunks  # type: ignore
+            # to compensate.
+            self.scale_momentum(nchunks)
+
         normalized_chunks = [self.batchnorm(chunk) for chunk in input.chunk(nchunks, 0)]
-        if has_momentum:
-            self.batchnorm.momentum = original_momentum  # type: ignore
+
+        if self.training and scale:
+            self.unscale_momentum(original_momentum)
 
         return torch.cat(normalized_chunks, dim=0)
 
     @staticmethod
-    def from_batchnorm(module: torch.nn.Module, ghost_batch_size: int) -> _GhostBatchNorm:
+    def from_batchnorm(module: torch.nn.Module, ghost_batch_size: int) -> "_GhostBatchNorm":
         assert isinstance(module, _TORCH_BATCHNORM_BASE_CLASS), "Module is not a BatchNorm subclass!"
         bn_type = _corresponding_ghost_batchnorm_type(module)
         return bn_type(ghost_batch_size=ghost_batch_size, base_batchnorm=module)
 
+    @torch.jit.unused
```

Comment:
To support `ghost_batchnorm` in `eval` mode, we refactor the offending lines into functions. Torchscript does not support accessing the `momentum` attribute, a pytorch issue has been filed.